### PR TITLE
osquery: revision for boost

### PR DESCRIPTION
**Pull request by ilovezfs**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

**Comment by ilovezfs** [comment]
Reported upstream https://github.com/facebook/osquery/issues/3598.

**Comment by theopolis** [comment]
Maybe a solution: https://stackoverflow.com/questions/31665095/xcode-error-compiling-c-expected-member-name-or-after-declaration-specifie

I can try to apply this to osquery's master, if we pull in `boost` 1.65 we should be able to reproduce.

**Comment by ilovezfs** [comment]
Thanks, @theopolis!

File: Formula/osquery.rb [hunk 1]
```diff
@@ -77,8 +78,23 @@ class Osquery < Formula
     sha256 "bf85b2d805f7cd7c4bc0c618c756b02ce618e777b727041ab75197592c4043f2"
   end
 
+  # upstream: 'deps: Update boost to 1.65'
+  # https://github.com/facebook/osquery/pull/3600
+  patch do
+    url "https://github.com/facebook/osquery/commit/70fe4ab194e.patch?full_index=1"
```
Comment (theopolis) [inline Formula/osquery.rb]: Since you're already using `boost` 1.65, you shouldn't need to pull in this PR.

**Review by theopolis — CHANGES_REQUESTED** [review]

File: Formula/osquery.rb (hunk 1 above)
Comment (JCount) [reply Formula/osquery.rb]: .... Good point ...

**Comment by JCount** [comment]
OK, that is essentially 🍏, the indicated failure is inconsequential.

**Comment by JCount** [comment]
🎊

**Pull request closed without merging**